### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by gnunicorn**
refs #4753

This:

- [x] removes git-dependency of wasmtime, switches to released substrate-wasmtime
- [x] update fdlimit to 0.1.4
- [x] update parity-common dependencies (waiting for crates.io releases)
- [x] update to latest substrate-bip39
- [x] update `schnorrkel` & `merlin`
- [x] cargo update
- [x] primitive-types (mismatch through `evm-*` modules): https://github.com/sorpaas/rust-evm/pull/3

### CODE REVIEW

File: bin/utils/subkey/Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ rand = "0.7.2"
 clap = "2.33.0"
 tiny-bip39 = "0.7"
 rustc-hex = "2.0.1"
-substrate-bip39 = "0.3.1"
+substrate-bip39 = "0.4.1"
```
Comment (NikVolf) [inline bin/utils/subkey/Cargo.toml]: This won't work, `substrate-bip39` 0.4 uses new schnorrkel which is not yet adapted for substrate properly

Comment (gnunicorn) [reply bin/utils/subkey/Cargo.toml]: > not yet adapted for substrate

What does that mean exactly? Do you mean on types? We had some issues with marlin and evm regarding then using different types, but that was all resolved.

Or do you mean the underlying algorithm has changed? If so, where can I read up about that, the crate doesn't have any changelog :( .

Comment (NikVolf) [reply bin/utils/subkey/Cargo.toml]: Ok, seems that schnorrkel get updated to version 0.9.1 (was 0.9.0 recently) which is compatible with current substrate

So I take my words back

File: primitives/core/Cargo.toml [hunk 2]
```diff
@@ -33,14 +33,14 @@ parking_lot = { version = "0.10.0", optional = true }
...
-parity-util-mem = { version = "0.5.2", default-features = false, features = ["primitive-types"] }
+parity-util-mem = { version = "0.6.0", default-features = false, features = ["primitive-types"] }
 futures = { version = "0.3.1", optional = true }
 
 # full crypto
 ed25519-dalek = { version = "1.0.0-pre.3", default-features = false, features = ["u64_backend", "alloc"], optional = true }
 blake2-rfc = { version = "0.2.18", default-features = false, optional = true }
 tiny-keccak = { version = "2.0.1", features = ["keccak"], optional = true }
-schnorrkel = { version = "0.8.5", features = ["preaudit_deprecated", "u64_backend"], default-features = false, optional = true }
+schnorrkel = { version = "0.9", features = ["preaudit_deprecated", "u64_backend"], default-features = false, optional = true }
```
Comment (bkchr) [inline primitives/core/Cargo.toml]: This is not correct, we need at least `0.9.1`

**Comment by bkchr** [comment]
Can we please stop merging prs without any approval? Especially the schnorrkel change could break Kusama.

**Comment by gnunicorn** [comment]
> Can we please stop merging prs without any approval?

Sorry about that, wasn't meant to become a habit. Wanted to merge a different PR and didn't look closer before pressing the button. Will do better in the future (and not merge on low sugar again!)

**Pull request merged**